lib/src/util/side_effect_visitor.dart

```diff
@@ -0,0 +1,546 @@
+import 'package:analyzer/analyzer.dart';
+
+class SideEffectVisitor extends RecursiveAstVisitor {
+  var _hasSideEffect = false;
+
+  bool get hasSideEffect => _hasSideEffect;
+  @override
+  visitAdjacentStrings(AdjacentStrings node) {
+    super.visitAdjacentStrings(node);
```

Comment:
Exactly, that is why I said that I can remove them all, it is just to check all the types one by one.